### ISSUE TITLE
Add public preview for text files

### ISSUE THREAD

**Issue by PVince81** [owncloud/files_texteditor#13]
### Steps to reproduce:
1. Create text file
2. Share as public link
3. Open public link
### Actual result

Ugly preview (enlarged thumbnail)
### Expected result

Show the actual text, maybe with a white background.
Keep it read-only as a first step.

**Comment by oparoz** [comment]
https://github.com/owncloud/core/issues/15652 implements a text preview of text files which are publicly shared.
This can always be improved later by loading the content inside the text editor in read-only mode so that it looks better(, as in formatted).

**Comment by luckydonald** [comment]
From what I understand the current state is to display the first 1000 characters in a text div.

But now there is a [preview api in the text editor](https://github.com/owncloud/files_texteditor/blob/5419c95d207e4459161a7b91709b03f981878c92/README.md#preview-apps), right?
I think, the preview should be the preview on shared links as well.
I would consider that as the expected result.
That would also benefit plugins like the [markdown plugin](https://github.com/icewind1991/files_markdown) which are already implementing a neat preview in the editor.

(Not sure if I should open a new issue for this...)

**Comment by oparoz** [comment]
>But now there is a preview api in the text editor, right?

Not from what I understand from that page. It's now possible to extend `text_editor` with more parser, but there is no API to return a parsed preview of a file

**Comment by tomneedham** [comment]
> It's now possible to extend text_editor with more parsers, but there is no API to return a parsed preview of a file

Correct. To get public viewing of files working we need to patch the editor to use the default endpoints for returning a file.

**Comment by luckydonald** [comment]
Not knowing the internals, *but* why do we need changes in the editor?
Would we not only need the download page call the plugin's parser as well?

**Comment by tomneedham** [comment]
So right now the only way for 'preview' plugins on the public page to access files is for them to access the file in question using `context.fileList.getDownloadUrl(file, dir)` as Vincent mentioned above. Again many code examples have even been posted above. Pull requests are always welcomed, just right now I dont have the time to work on this.
